packages/datasource-elasticsearch/test/crud/aggregate.integration.test.ts

```diff
@@ -0,0 +1,368 @@
+import { Aggregation, Caller, Filter } from '@forestadmin/datasource-toolkit';
+
+import { getCollectionForIndex } from '../helpers/elastic-search-collection';
+import {
+  createElasticsearchIndex,
+  deleteElasticsearchIndex,
+} from '../helpers/elastic-search-index-manager';
+
+const dataset = [
+  {
+    discriminant: 'blue',
+    weight: 1,
+    date: '2023-11-23',
+  },
+  {
+    discriminant: 'blue',
+    weight: 21,
+    date: '2023-11-21',
+  },
+  {
+    discriminant: 'blue',
+    weight: 3,
+    date: '2023-10-21',
+  },
+  {
+    discriminant: 'red',
+    weight: 4,
+    date: '2023-06-21',
+  },
+  {
+    discriminant: 'red',
+    weight: 5,
+    date: '2023-01-21',
+  },
+  {
+    discriminant: 'red',
+    weight: 6,
+    date: '2022-01-01',
+  },
+];
+
+const indexName = 'test-crud-update';
+
+beforeAll(async () => {
+  await createElasticsearchIndex(indexName, dataset);
+});
+
+afterAll(async () => {
+  await deleteElasticsearchIndex(indexName);
+});
+
+describe('Collection > CRUD > update', () => {
+  describe('Count', () => {
+    it('should count record record', async () => {
+      const collection = await getCollectionForIndex(indexName);
+
+      const result = await collection.aggregate(
+        null as unknown as Caller,
+        {} as unknown as Filter,
+        { operation: 'Count' } as unknown as Aggregation,
+      );
+
+      expect(result[0]).toMatchObject(expect.objectContaining({ value: 6 }));
+    });
+
+    it('should count record record with filter', async () => {
+      const collection = await getCollectionForIndex(indexName);
+
+      const result = await collection.aggregate(
+        null as unknown as Caller,
+        {
+          conditionTree: {
+            field: 'discriminant',
+            operator: 'Equal',
+            value: 'red',
+          },
+        } as unknown as Filter,
+        { operation: 'Count' } as unknown as Aggregation,
+      );
+
+      expect(result[0]).toMatchObject(expect.objectContaining({ value: 3 }));
+    });
+
+    it.skip('should count aggregate', async () => {
```

Comment:
Why do skip this one ?

Reply:
Because it's broken due to `discriminant` being of type `text` and not `keyword`.

Reply:
We need to be able to specify the mapping when creating the index.